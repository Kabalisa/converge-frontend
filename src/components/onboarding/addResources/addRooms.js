import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import { Grid } from 'semantic-ui-react';
import * as styled from './customStyles';
import { GET_SPECIFIC_ROOMS } from '../../../graphql/queries/Rooms';


const AddRooms = () => {
  const {
    data: { allRooms },
    loading,
  } = useQuery(GET_SPECIFIC_ROOMS);

  if (loading) return <p>Loading...</p>;

  return (
    <div>
      <styled.GridContainer>
        <Grid.Column mobile={16} tablet={8} computer={8}>
          <styled.RoomImgOne src={allRooms.rooms[0].imageUrl} width="180" />
          <styled.centerImageText>
            <styled.centerImageTextWhiteUp>
              {allRooms.rooms[0].name}
            </styled.centerImageTextWhiteUp>
            <styled.centerImageTextTwoDown />
          </styled.centerImageText>
        </Grid.Column>

        <Grid.Column mobile={16} tablet={8} computer={8}>
          <styled.RoomImgTwo width="180" src={allRooms.rooms[1].imageUrl} />
          <styled.centerImageText>
            <styled.centerImageTextTwo>
              {allRooms.rooms[1].name}
            </styled.centerImageTextTwo>
            <styled.centerImageTextTwoDown />
          </styled.centerImageText>
        </Grid.Column>
      </styled.GridContainer>
    </div>
  );
};


export default AddRooms;
