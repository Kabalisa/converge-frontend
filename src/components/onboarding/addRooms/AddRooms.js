/* eslint-disable import/no-unresolved */
/* eslint-disable jsx-a11y/alt-text */
import React, { Component, Fragment } from 'react';
import '../../../assets/styles/addroom.scss';
import BuildingCardFloor from '../../commons/BuildingCardFloor';
import InputWithNumbers from '../../commons/InputWithNumbers';
import roomImage from '../../../assets/images/roomImage.png';
import roomImage2 from '../../../assets/images/roomImage2.png';
import { Input } from '../../commons/Input';
import backIcon from '../../../assets/images/ic_back.svg';


class AddRooms extends Component {
  state= {
    Floors: ['Floor 1', 'Floor 2'],
    activeFloor: 'Floor 1',
    block: 'Block A',
  };
  render() {
    const { Floors, activeFloor, block } = this.state;
    return (
      <Fragment>
        <div>
          <div className="addroom-page-title">
            <div className="back_icon">
              <img src={backIcon} alt="back icon" />
            </div>
            <div>
              <span className="addroom-title">Add Rooms</span>
            </div>
          </div>
          <div>
            <p>Set the structure of your Center</p>
            <BuildingCardFloor Floors={Floors} activeFloor={activeFloor} block={block} />
          </div>
          &nbsp;&nbsp;&nbsp;&nbsp;
        </div>
        <div>
          <p>How many meetings rooms are in Floor 1?</p>
          <InputWithNumbers />
        </div>
        &nbsp;&nbsp;&nbsp;&nbsp;
        <div>
          <p>Name Your Meetings Rooms</p>
          <div className="name-rooms">
            <div className="name-rooms-left">
              <div className="name-rooms-image">
                <img src={roomImage} alt="room name" />
                <Input placeholder="Congitio" />
              </div>
            </div>
            <div className="name-rooms-right">
              <div className="name-rooms-image">
                <img src={roomImage2} alt="room name" />
                <Input placeholder="Congitio" />
              </div>
            </div>

          </div>
        </div>
      </Fragment>
    );
  }
}
export default AddRooms;
