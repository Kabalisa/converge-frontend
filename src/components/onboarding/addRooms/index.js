import React from 'react';
import PropTypes from 'prop-types';
import AddRooms from './AddRooms';
import OnboardingLayout from '../../../containers/OnboardingLayout';


export const AddroomsDesign = ({ handleOnClick }) => (
  <OnboardingLayout
    layoutLeft={<AddRooms handleOnClick={handleOnClick} />}
    layoutRight={<p> the preview of your setup</p>}
  />
);

AddroomsDesign.propTypes = {
  handleOnClick: PropTypes.func.isRequired,
};

export default AddroomsDesign;
