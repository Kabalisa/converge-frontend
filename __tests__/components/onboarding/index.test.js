import React from 'react';
import { shallow } from 'enzyme';
import { AddroomsDesign } from '../../../src/components/onboarding/addRooms/index';


const props = {
  onBoarding: jest.mock('../../../src/containers/OnboardingLayout', () => ({
    layoutLeft: jest.fn(),
    layoutRight: '<p>Hello bro</p>',
  })),
};

describe('Add room design', () => {
  const component = shallow(<AddroomsDesign {...props} />);
  it('should render with succesfully', () => {
    expect(component).toMatchSnapshot();
  });
});
