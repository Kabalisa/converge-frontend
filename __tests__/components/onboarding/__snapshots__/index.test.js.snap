// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add room design should render with succesfully 1`] = `
<OnboardingLayout
  layoutLeft={
    <AddRooms
      handleOnClick={undefined}
      setFloors={Array []}
      setactiveFloor=""
    />
  }
  layoutRight={
    <p>
       the preview of your setup
    </p>
  }
/>
`;
